conman/routes/handlers.py

```diff
@@ -23,11 +24,48 @@ def check(cls, route):
         return []
 
     def handle(self, request, path):
-        """Raise an error if a subclass calls handle without defining how.."""
+        """Raise an error if a subclass calls handle without defining how."""
         msg = 'Subclasses of `BaseHandler` must implement `handle()`.'
         raise NotImplementedError(msg)
 
 
+class TemplateHandler(BaseHandler):
+    """
+    A handler that renders a template.
+
+    Routes using this handler must define `template_name`, which can either
+    be a string or list, as accepted by `django.shortcuts.render`.
+
+    Renders the `template_name` of the related Route into an
+    `HttpResponse`.
+    """
+    @classmethod
+    def check(cls, route):
+        """Ensure route has a template_name attribute."""
+        if not hasattr(route, 'template_name'):
+            return [checks.Warning(
+                '{} must have a `template_name` attribute.'.format(route.__name__),
+                hint=('This is a requirement of {}.'.format(cls.__name__)),
+                obj=route,
+            )]
+
+        return []
+
+    def handle(self, request, path):
+        """
+        Render a template and return the `HttpResponse` returned.
+
+        Raises `django.core.urlresolvers.Resolver404` if `path` isn't found.
+        """
+        if path != '/':
```

Comment:
Why do we expect `path == '/'`?

Reply:
The `path` represents what's left of the URL that was browsed to once `route.url` is subtracted.

For example, if the only `Route` in the DB is this one...

```python
route = Route.objects.create(url='/articles/')
```

...then when we browse to `/articles/latest/`, this method would be called with `url='/latest/', and `self.route` would be `route`.

The `TemplateHandler`, like the `ViewHandler`, does not know how to deal with subpaths in the url, so it makes sense to raise a 404. (The `URLConfHandler` does have the ability to deal with these subpaths.)